packages/mc-scripts/src/utils/get-config-diff.js

```diff
@@ -0,0 +1,252 @@
+const { red, green } = require('chalk');
+
+const keysToSkip = new Set(['id']);
+
+const t = (indentLevel) => '  '.repeat(indentLevel);
+
+const getStringDiff = (oldStr, newStr, label, indentLevel = 0) => {
+  if (!oldStr && newStr) {
+    return `${t(indentLevel)}${label} added: ${green(newStr)}`;
+  } else if (oldStr && !newStr) {
+    return `${t(indentLevel)}${label} removed: ${red(oldStr)}`;
+  } else if (oldStr && newStr && oldStr !== newStr) {
+    return `${t(indentLevel)}${label} changed: ${red(oldStr)} => ${green(
+      newStr
+    )}`;
+  }
+};
+
+const getArrayDiff = (oldArray, newArray, indentLevel = 0) => {
+  const oldArraySet = new Set(oldArray);
+  const arrayDiff = [];
+
+  newArray?.forEach((item) => {
+    if (oldArraySet.has(item)) {
+      oldArraySet.delete(item);
+    } else {
+      arrayDiff.push(`${t(indentLevel)}Item with "${green(item)}" added`);
+    }
+  });
+  oldArraySet.forEach((item) => {
+    arrayDiff.push(`${t(indentLevel)}Item with "${red(item)}" removed`);
+  });
+
+  return arrayDiff.join('\n');
+};
+
+const getPermissionsDiff = (oldPermissions, newPermissions) => {
+  const permissionDiff = ['permissions changed'];
+  const mappedOldPermission = {};
+
+  oldPermissions.forEach(({ name, oAuthScopes }) => {
+    mappedOldPermission[name] = oAuthScopes;
+  });
```

Comment:
nit: Perhaps we could use `reduce` here instead of `forEach`.
```suggestion
const mappedOldPermissions = oldPermissions.reduce(
  (acc, { name, oAuthScopes }) => ({
    ...acc,
    [name]: oAuthScopes,
  }),
  {}
);
```

Reply:
Updated.